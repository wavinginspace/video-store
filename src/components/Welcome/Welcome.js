import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import "./Welcome.scss"

class Welcome extends Component {
  render() {
    return (
      <nav className="Welcome box">
        <Link to="/login">
          <button className="welcome button">Login</button>
        </Link>
        <Link to="/register">
          <button className="welcome button">Register</button>
        </Link>
      </nav>
    );
  }
}

export default Welcome;
