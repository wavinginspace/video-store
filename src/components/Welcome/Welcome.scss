@import '../../styles/_breakpoints.scss';
@import '../../styles/_variables.scss';

.Welcome {
  text-align: center;
}

button {
  display: block;
  padding: 1rem;
  text-decoration: none;
}

.demo-link {
  text-decoration: none;
  color: var(--main);
  margin: 0 auto;
  text-align: center;
  display: block;
  font-weight: 500;
  transition: color 0.2s;
}

.demo-link:hover {
  color: blue;
  font-weight: 500;
}

.login,
.register {
  text-align: center;
  background-color: var(--green);
  text-decoration: none;
  margin: 2rem auto;
  display: block;
  padding: 0.8rem;
  width: 200px;
  font-size: 1rem;
  font-weight: 500;
  transition: color 0.2s;
}

.login-link,
.register-link {
  text-decoration: none;
}

.login:hover,
.register:hover {
  color: white;
  background-color: var(--buttonhov);
}

.welcome-message {
  margin-bottom: 1.5rem;
}
