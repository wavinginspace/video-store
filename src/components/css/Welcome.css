:root {
  --main: var(--main);
  --nav: #499acc;
  --headergrad: linear-gradient(0deg, #ff7676, #fffd8f);
  --light: #fffd8f;
  --headerog: #ff7676;
}

.Welcome {
  text-align: center;
}

button {
  display: block;
  padding: 1rem;
}
/*# sourceMappingURL=Welcome.css.map */