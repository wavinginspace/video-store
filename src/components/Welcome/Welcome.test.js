import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import Welcome from './Welcome';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(
    <Router>
      <Welcome />
    </Router>,
    div
  );

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <Router>
        <Welcome />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
