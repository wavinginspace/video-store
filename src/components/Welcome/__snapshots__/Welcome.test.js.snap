// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<nav
  className="Welcome box"
>
  <a
    href="/login"
    onClick={[Function]}
  >
    <button
      className="welcome button"
    >
      Login
    </button>
  </a>
  <a
    href="/register"
    onClick={[Function]}
  >
    <button
      className="welcome button"
    >
      Register
    </button>
  </a>
</nav>
`;
